numpyro/infer/mcmc.py

```diff
@@ -382,11 +383,76 @@ def _compile(self, rng_key, *args, extra_fields=(), init_params=None, **kwargs):
         except TypeError:
             pass
 
+    @property
+    def init_state(self):
+        """
+        The initial state to run MCMC. If this attribute is not None,
+        :meth:`run` will skip calling `self.sampler.init(...)` method to
+        obtain the initial state.
+
+        .. warning:: If you want to set a value for this attribute, then making sure
+            that some fields such as `HMCState.i`, `HMCState.mean_accept_prob` are
+            reseted to 0. In addition, if the initial state has invalid fields such as
+            `HMCState.potential_energy` (because e.g. `data` of the new `run` has been
+            changed), the sampler might get trouble during some initial MCMC steps.
+            To resolve that, we can set `potential_energy` to a high value (says `1e38`),
+            so that it is more likely to accept a new proposal (which will have a correct
+            `potential_energy` field) in the first MCMC step.
+            For example
+
```

Comment:
nit:
If you want to set a value for this attribute, then you will need to make sure
that certain fields such as `HMCState.i` and `HMCState.mean_accept_prob` are
reset to 0. In addition, if any of the fields of the initial state have invalid values 
(e.g. `HMCState.potential_energy` may be invalid if `data` of the new `run` has been
changed), the sampler might get stuck or otherwise perform suboptimally.
To resolve this issue, you can set `potential_energy` to a large value (say `1e38`),
so that the sampler is more likely to accept a new proposal in the first MCMC step 
(and which will have a correct `potential_energy`).